### ISSUE TITLE
Add a ProcessMonitor plugin for twistd

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4356]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)| @wallrj reported|
|-|-|
|Trac ID|trac#4356|
|Type|enhancement|
|Created|2010-03-07 15:15:47Z|
|Branch|https://github.com/twisted/twisted/tree/procmon-plugin-4365|

It would be useful to be able to launch t.r.procmon.ProcessMonitor from the command line as a simple way to monitor and automatically restart another (unreliable) process. eg
```
twistd procmon squid -N
```

A plugin would also provide a nice example of procmon use and replace the ugly __main__ function currently at the bottom of the module.

Attachments:

* [procmon-plugin-4365.1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/671/671cecdd43a2b519e266592124a057e7dde49eee/4636400a1afbd2ec9f915339cd13cba01070a11b.patch) (6548 bytes) - added by rwall on 2010-05-15 16:01:21Z - A procmon plugin and tests.

<details><summary>Searchable metadata</summary>

```
trac-id__4356 4356
type__enhancement enhancement
reporter__rwall rwall
priority__normal normal
milestone__ 
branch__branches_procmon_plugin_4365 branches/procmon-plugin-4365
branch_author__rwall rwall
status__closed closed
resolution__fixed fixed
component__runner runner
keywords__ 
time__1267974947000000 1267974947000000
changetime__1275803371000000 1275803371000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_1"></a>|@wallrj removed owner|
|-|-|

Ready for review in [procmon-plugin-4365.1.patch](None/procmon-plugin-4365.1.patch) 
 * NB this patch is against [branches/parameterize-procmon-3775-2](branches/parameterize-procmon-3775-2)

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_2"></a>|@wallrj commented|
|-|-|

Looks like this isn't a new idea. Tv added a ProcMon plugin in 2003 which was later removed by moshez:
 * r5715
 * r8721

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

Applied patch to a branch so we can see [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/procmon-plugin-4365).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph commented|
|-|-|

Gave the branch the wrong number, unfortunately.  I'll fix it if I merge forward.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli set owner to @cyli|
|-|-|

The tests pass, and running `twistd procmon sleep 10` works - `sleep 10` is restarted over and over.

Made two small changes:
 1.  Gave t.r.test.test_procmontap.ProcessMonitorTapTest a class (unittest.TestCase) level docstring. 
 1.  Reflowed some of the text in t.r.procmontap.py's and t.r.test.test_procmontap.py's lines wrapped to 79 columns.  Also put in extra newlines so that t.r.procmontap.py has the right number of newlines between methods/class and methods.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli set status to `closed`|
|-|-|

(In [29212]) Merge procmon-plugin-4365: Add a ProcessMonitor plugin for twistd 

Author: rwall 
Reviewer: cyli 
Fixes: [#4356](https://github.com/twisted/twisted/issues/4356)

Added a twistd plugin to launch twisted.runner.procmon.ProcessMonitor from the command line.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
